nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-shared/src/main/java/org/apache/nifi/kafka/shared/aad/CustomAuthenticateCallbackHandler.java

```diff
@@ -0,0 +1,87 @@
+package org.apache.nifi.kafka.shared.aad;
+
+import com.microsoft.aad.msal4j.*;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.security.auth.AuthenticateCallbackHandler;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerTokenCallback;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.AppConfigurationEntry;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeoutException;
+
+public class CustomAuthenticateCallbackHandler implements AuthenticateCallbackHandler {
```

Comment:
CustomAuthenticateCallbackHandler should be renamed to AADAuthenticateCallbackHandler as the authorizer is only support https://login.microsoftonline.com/